includes/orders/functions/orders.php

```diff
@@ -1375,42 +1375,125 @@ function edd_get_net_order_statuses() {
+	return apply_filters( 'edd_complete_order_status_keys', $statuses );
 }
 
 /**
- * Generate unique payment key for orders.
+ * Get the order statuses which are considered completed.
  *
  * @since 3.0
- * @param string $key Additional string used to help randomize key.
- * @return string
+ * @return array An array of order statuses which are considered completed.
  */
-function edd_generate_order_payment_key( $key ) {
-	$auth_key    = defined( 'AUTH_KEY' ) ? AUTH_KEY : '';
-	$payment_key = strtolower( md5( $key . gmdate( 'Y-m-d H:i:s' ) . $auth_key . uniqid( 'edd', true ) ) );
+function edd_complete_order_statuses() {
+	return array_intersect_key( edd_get_payment_statuses(), array_flip( edd_complete_order_status_keys() ) );
+}
+
+/**
+ * Get the order status array keys which are considered incomplete.
+ *
+ * @since 3.0
+ * @return array An array of order status keys which are considered incomplete.
+ */
+function edd_incomplete_order_status_keys() {
+	$statuses = array( 'pending', 'abandoned', 'processing', 'failed' );
 
 	/**
-	 * Filters the payment key
+	 * Order statuses which are considered incomplete.
+	 *
+	 * @param $statuses {
+	 *        An array of order status array keys.
+	 * }
+	 */
+	return apply_filters( 'edd_incomplete_order_status_keys', $statuses );
+}
+
+/**
+ * Get the order statuses which are considered incomplete.
+ *
+ * @since 3.0
+ * @return array An array of order statuses which are considered incomplete.
+ */
+function edd_incomplete_order_statuses() {
+	return array_intersect_key( edd_get_payment_statuses(), array_flip( edd_incomplete_order_status_keys() ) );
+}
+
+/**
+ * Get the order status array keys which are considered recoverable.
+ *
+ * @since 3.0
+ * @return array An array of order status keys which are considered recoverable.
+ */
+function edd_recoverable_order_status_keys() {
+	$statuses = array( 'pending', 'abandoned', 'failed' );
+
+	/**
+	 * Order statuses which are considered recoverable.
+	 *
+	 * @deprecated This has been deprecated in EDD 3.0. Use edd_recoverable_order_statuses instead.
+	 * @param $statuses {
+	 *        An array of order status array keys.
+	 * }
+	 */
+	$statuses = apply_filters( 'edd_recoverable_payment_statuses', $statuses );
+
+	/**
+	 * Order statuses which are considered recoverable.
 	 *
 	 * @since 3.0
-	 * @param string $payment_key The value to be filtered
-	 * @param string $key Additional string used to help randomize key.
-	 * @return string
+	 * @param $statuses {
+	 *        An array of order status array keys.
+	 * }
 	 */
-	return apply_filters( 'edd_generate_order_payment_key', $payment_key, $key );
+	return apply_filters( 'edd_recoverable_order_status_keys', $statuses );
+}
+
+/**
+ * Get the order statuses which are considered non-completed.
+ *
+ * @since 3.0
+ * @return array An array of order statuses which are considered not completed.
+ */
+function edd_recoverable_order_statuses() {
```

Comment:
Because this changes the array to include labels, it's breaking the order `is_recoverable` method. I am very strongly inclined to not have `_keys` methods.

Reply:
Note: we have no tests covering the `is_recoverable` method and we probably should.